lib/filesystem.js

```diff
@@ -112,6 +112,8 @@ FileSystem.prototype.getItem = function(filepath) {
     if (item) {
       if (item instanceof Directory && name !== currentParts[i]) {
         // make sure traversal is allowed
+        // This fails for Windows directories which do not have execute permission, by default. It may be a good idea
```

Comment:
The existing behaviour here might be of concern, later. Is there a better way to accommodate Windows?

I've added a workaround which can be seen [here](https://github.com/tschaub/mock-fs/pull/304/files#diff-6d186b954a58d5bb740f73d84fe39073R260).